File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::placeholders;

class smartphoneNode : public rclcpp::Node {
    public:
        smartphoneNode() : Node("smartphone") {
            subscriber_ = this->create_subscription<example_interfaces::msg::String>(
                "robot_news", 10, 
                std::bind(&smartphoneNode::callbackRobotNews, this, _1));
            RCLCPP_INFO(this->get_logger(), "Smartphone started");
        }

    private:
        void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg) {
            RCLCPP_INFO(this->get_logger(), "I heard '%s'", msg->data.c_str());
        } 

        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<smartphoneNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env pyhton3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class smartphoneNode(Node):
    def __init__(self):
            super().__init__("smartphone")
            self.subscriber_ = self.create_subscription(
                  String, "robot_news", self.callback_robot_news, 10)
            self.get_logger().info("Start listening robot_news topic")

    def callback_robot_news(self, msg: String):
          self.get_logger().info("i heard \'" + msg.data + "\'")

def main(args=None):
      rclpy.init(args=args)
      node = smartphoneNode()
      rclpy.spin(node)
      rclpy.shutdown()


if __name__ == "__main__":
      main()